linkerd/http-route/src/http/match/path.rs

```diff
@@ -102,6 +102,36 @@ impl std::cmp::Ord for PathMatch {
     }
 }
 
+#[cfg(feature = "proto")]
+pub mod proto {
+    use super::*;
+    use linkerd2_proxy_api::http_route as api;
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum InvalidPathMatch {
+        #[error("missing match")]
+        MissingMatch,
+
+        #[error("invalid regular expression: {0}")]
+        Regex(#[from] regex::Error),
+    }
+
+    // === impl MatchPath ===
+
+    impl TryFrom<api::PathMatch> for MatchPath {
+        type Error = InvalidPathMatch;
+
+        fn try_from(rm: api::PathMatch) -> Result<Self, Self::Error> {
+            // TODO parse paths to validate they're valid.
```

Comment:
do we want to address this now, or is it something we're going to do later?

Reply:
See TODOs in PR description. I'm not sure it's required.